### PR TITLE
WIP Changing name of PySolverInterface to precice

### LINKED ISSUE #265: Rename PySolverInterface
I think the name `PySolverInterface` is a bad name. It is not really specific and we should find a better name. If you have any propositions, feel free to comment below.

# Consequences

We will have to change all references to the `PySolverInterface` module. Here is an incomplete list of tutorials/solvers... using `PySolverInterface`:

* https://github.com/precice/elastictube1d/tree/master/PythonTube
* https://github.com/precice/fenics-adapter
* feel free to add any missing items to this list

### PR DESCRIPTION
**Pull request by BenjaminRueth**
resolves #265. The wrapper module `PySolverInterface` avoids breaking the API. We have to update the documentation correspondingly.

### CODE REVIEW

**Comment by BenjaminRueth** [comment]
Before we merge this PR to `develop`, we should merge https://github.com/precice/precice/pull/274

File: src/precice/bindings/PySolverInterface/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+*Note:* If you installed Cython using `apt install Cython` under Ubuntu 16.04, you might face problems. Try installing using `pip2/3` instead.
+
+# Building
+
+1. Install the new language bindings `precice` following the instructions [here](https://github.com/precice/precice/tree/changingNameOfPySolverInterface/src/precice/bindings/python).
+2. Open terminal in this folder.
+3. Execute the following command:
+
+```
+$ python setup.py build
```
Comment (fsimonis) [inline src/precice/bindings/PySolverInterface/README.md]:
```suggestion
$ python setup.py build_ext --include-dirs=$PRECICE_ROOT/src --library-dirs=$PRECICE_ROOT/build/last
```

File: src/precice/bindings/PySolverInterface/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+```
+This creates a folder `build` with the binaries.
```
Comment (fsimonis) [inline src/precice/bindings/PySolverInterface/README.md]:
```suggestion
This creates a folder `build` with the binaries.
If you build preCICE using CMake, you can pass the path to the CMake binary directory using `--library-dirs`.
```

File: src/precice/bindings/PySolverInterface/PySolverInterface/PySolverInterface.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
+from precice import Interface
```
[the remark this answers is not in the document]
Comment (BenjaminRueth) [reply src/precice/bindings/PySolverInterface/PySolverInterface/PySolverInterface.py]: Done: https://github.com/precice/precice/commit/a801832e3e4e1d08f91c052648773da93e966b2b

**Comment by BenjaminRueth** [comment]
if we are using `--include-dirs=$PRECICE_ROOT/src --library-dirs=$PRECICE_ROOT/build/last` this also resolves #220.

**Pull request merged**